test/jdk/java/io/BufferedInputStream/TransferToTrusted.java

```diff
@@ -58,38 +59,31 @@ public void write(int b) throws IOException {
         }
     }
 
-    public static void main(String[] args) throws IOException {
+    public static void main(String[] args) throws Exception {
         final int length = 128;
         byte[] buf = new byte[length];
         RND.nextBytes(buf);
-        byte[] dup = Arrays.copyOf(buf, buf.length);
 
-        var bis = new BufferedInputStream(new ByteArrayInputStream(dup));
-        bis.mark(dup.length);
+        var outputStreams = new OutputStream[]{
+                new ByteArrayOutputStream(),
+                new FileOutputStream(File.createTempFile(TransferToTrusted.class.getName(), null)),
+                new PipedOutputStream(new PipedInputStream(length)),
+                new UntrustedOutputStream()
+        };
 
-        var tempFile = File.createTempFile(TransferToTrusted.class.getName(), null);
-        try {
-            var outputStreams = new OutputStream[]{
-                    new ByteArrayOutputStream(),
-                    new FileOutputStream(tempFile),
-                    new PipedOutputStream(new PipedInputStream(length)),
-                    new UntrustedOutputStream()
-            };
-
-            for (var out : outputStreams) {
-                System.err.println("out: " + out.getClass().getName());
+        for (var out : outputStreams) {
+            System.err.println("out: " + out.getClass().getName());
 
+            var bis = new BufferedInputStream(new ByteArrayInputStream(buf.clone()));
+            try (out; bis) {
+                bis.read();//need this to fill the BIS.buf in
                 bis.transferTo(out);
-                bis.reset();
-                try (out) {
-                    if (!Arrays.equals(buf, bis.readAllBytes())) {
-                        throw new RuntimeException("Internal buffer was modified");
-                    }
+                var internalBuffer = bis.getClass().getDeclaredField("buf");
+                internalBuffer.setAccessible(true);
+                if (!Arrays.equals(buf, Arrays.copyOf((byte[]) internalBuffer.get(bis), length))) {
```

Comment:
This is not testing the absence of a copy.

Reply:
Yeah, but how could I intercept the argument of OutputStream.write() and check it's identity?

Reply:
As I wrote earlier today to Vladimir (see above) I actually do not see a solution for this, frankly spoken, as the interceptor would not be trusted. That's why I asked Brian for his ok to keep the test as-is. We need to wait for his judgement, as he is the reviewer, not me (I only try to give initial advice). My comment was more a hint for me to wait for Brian's judgement, less a request to you to fix this.

Reply:
@bplb are you ok with the current implementation and its test?